client/my-sites/domains/domain-management/dataviews/use-actions.tsx

```diff
@@ -0,0 +1,259 @@
+import { AutoRenewDiolog } from './components/auto-renew-dialog';
+import { useDomainsDataViewsContext } from './use-context';
+
+export function useActions( { sidebarMode }: { sidebarMode?: boolean }, onClose?: () => void ) {
+	const translate = useTranslate();
+	const {
+		getFullDomain,
+		sites,
+		getSiteSlug,
+		isAllSitesView,
+		domainStatusPurchaseActions,
+		updatingDomain,
+		userCanSetPrimaryDomains = false,
+		onDomainAction,
+		handleUpdateContactInfo,
+	} = useDomainsDataViewsContext();
+
+	const actions: Action< PartialDomainData >[] = [
+		{
+			id: 'manage-domain',
+			isPrimary: true,
+			icon: drawerLeft,
+			label: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				return domain && domain.type === domainTypes.TRANSFER
+					? translate( 'View transfer' )
+					: translate( 'View settings' );
+			},
+			callback: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				if ( ! domain ) {
+					return;
+				}
+				const url = domainManagementLink( domain, getSiteSlug( domains[ 0 ] ), isAllSitesView );
+				navigate( url );
+			},
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return domain.type !== domainTypes.WPCOM;
+			},
+		},
+		{
+			id: 'manage-dns-settings',
+			callback: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				if ( domain ) {
+					onDomainAction?.( 'manage-dns-settings', domain );
+					const url = domainManagementDNS( getSiteSlug( domains[ 0 ] ), domain.domain );
+					navigate( url );
+				}
+			},
+			label: translate( 'Manage DNS' ),
+			supportsBulk: false,
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return (
+					domain.canManageDnsRecords &&
+					domain.transferStatus !== transferStatus.PENDING_ASYNC &&
+					domain.type !== domainTypes.SITE_REDIRECT
+				);
+			},
+		},
+		{
+			id: 'manage-contact-info',
+			icon: <Icon icon={ info } />,
+			callback: ( domains: Array< PartialDomainData > ) => {
+				if ( domains.length === 0 ) {
+					return;
+				}
+				if ( domains.length === 1 ) {
+					const domain = domains[ 0 ];
+					const url = domainManagementEditContactInfo( getSiteSlug( domain ), domain.domain );
+					navigate( url );
+				} else {
+					handleUpdateContactInfo( domains );
+				}
+			},
+			label: translate( 'Manage contact information' ),
+			supportsBulk: ! sidebarMode,
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return (
+					domain.currentUserIsOwner &&
```

Comment:
This rule is not present in the existing action ([see](https://github.com/Automattic/wp-calypso/blob/trunk/packages/domains-table/src/domains-table/domains-table-row-actions.tsx#L68)). As a result, domains that have that action in production, don't have it in the dataviews-powered version.

Reply:
Yes, it is a bit complicated here.
It appeared here because it's also part of a bulk actions condition for the same action.
Initially, I didn't add it and I encountered cases when the action was available, but on the end screen I got a message that I'm not eligible for editing the contact information.
I just checked: the same happens with Domains Table.
![CleanShot 2025-01-24 at 23 07 10@2x](https://github.com/user-attachments/assets/cf77b35b-c955-4690-8441-ff23273f93ab)
So, I think having this condition here is a safer position. What do you think?